### ISSUE TITLE
FEATURE REQUEST: Hook for passing AWS API Gateway path parameters

### ISSUE THREAD

**Issue by kpeters-cbsi** [rayepps/exobase-js#2]
I'd like a hook that allows me to extract `{ path: 'foo/bar' }` from the path `/my/endpoint/foo/bar` where the endpoint is defined in the API as `/my/endpoint/{path}`. I've implemented it in my own project as follows, but I didn't want to submit a PR as I don't have tests defined for it and I don't know where it would be appropriate to put it in the codebase, as it's a hook but an API Gateway specific hook.

```
export { useJsonArgs, useQueryArgs, useHeaderArgs } from '@exobase/hooks';
import { partial } from 'radash';
import { useValidation } from '@exobase/hooks';
import { Props, ApiFunction, Request } from '@exobase/core';
import { APIGatewayProxyEvent } from 'aws-lambda';
import * as yup from 'yup';

type APIGatewayProxyRequest = Request & { event: APIGatewayProxyEvent };
type Yup = typeof yup;
type KeyOfType<T, Value> = { [P in keyof T]: Value };

/**
 * Get args from the parsed path parameters (AWS API Gateway-specific)
 */
export const useLambdaPathArgs = partial(
  useValidation,
  (props: Props<any, any, any, APIGatewayProxyRequest>) =>
    props.req.event.pathParameters
) as <TArgs = any>(
  shapeMaker: (yup: Yup) => KeyOfType<TArgs, any>
) => (func: ApiFunction) => ApiFunction;
```

**Comment by rayepps** [comment]
Interesting, I love the solution you've made in your own project! There is a big v2 coming with a ton of updates (including this one) next week 🎉 

I'll comment back when it's out.

**Comment by kpeters-cbsi** [comment]
> Interesting, I love the solution you've made in your own project! 

Thank you!

**Comment by rayepps** [comment]
Hey this is still in progress, I _think_ the work is done but its a lot of testing in different platforms haha

The latest versions are published as `v1.0.1-rc.x`, you can install them if you like until the actual `v1.0.0` is released. You can see the updated README and packages pr #3 (some of the packages were split up into more granular ones and/or renamed).

I'll comment back when v1.0.0 is published.
